include/boost/thread/detail/thread.hpp

```diff
@@ -560,7 +547,7 @@ namespace thread_detail {
         template<typename TimeDuration>
         inline bool timed_join(TimeDuration const& rel_time)
         {
-            return timed_join(get_system_time()+rel_time);
+            return do_try_join_until(detail::timespec_plus_internal_clock(rel_time));
```

Comment:
This has a sense only on Pthread. Windows has no timespec.

Reply:
Agreed. This particular instance is in an #else section that's only enabled when BOOST_THREAD_PLATFORM_WIN32 is not defined.

Reply:
RIght :)